### DISCUSSION TITLE
Saleor React Storefront + Saleor Checkout = 🚀

### DISCUSSION THREAD

**Discussion by karolkielecki** [saleor/saleor#10293] [Announcements]
Hi Saleors! 

One of our newest projects--Saleor Checkout--is now merged with Saleor React Storefront!

Explore our updated Saleor Demo! 
- Demo: https://demo.saleor.io/default-channel/en-US
- SRS repo: https://github.com/saleor/react-storefront

**Comment by tiendq** [comment]
I got this error when tried to check out.

> There was a problem
We couldn’t fetch information about your checkout. Go back to the store and try again.

**Reply by karolkielecki** [reply]
Hi @tiendq, did you receive that error on the demo?

**Reply by tiendq** [reply]
Yes, I just did a few steps:

1. Visit demo link above
2. Log in as admin (demo account)
3. Add a product to cart
4. Go to cart and click Checkout, then got above error

**Reply by karolkielecki** [reply]
Yeah, I've just received that error after adding one of the products to the basket and then tried to do checkout. We'll investigate it further and try to fix it asap. I'm sorry for the inconvenience. Thanks for reporting!

**Reply by typeofweb** [reply]
I was also able to reproduce it. Unfortunately it seems to be intermittent but we'll keep on digging. The issue is here: https://github.com/saleor/react-storefront/issues/345

**Reply by karolkielecki** [reply]
Hi @tiendq, we've provided a temporary fix to the demo. Thanks again for reporting!

**Comment by slamer59** [comment]
Hello, I began with saleor 3.1 ans develop a plugin for checkout. What steps should I make roughly to update to this version ?

Is this app managed by dashboard or ? Etc. Thanks for drawing the big lines.

**Reply by timuric** [reply]
You would have to update to the latest saleor core version to run checkout and storefront from this repo

**Reply by slamer59** [reply]
Ok. I will have a try.

**Comment by matteius** [comment]
I have been working to upgrade my saleor/react-storefront to this new pattern, and I would like clarification on something regarding payments vs transactions.   I have made some observations between what my currently old deployed react-storefront does compared to the new saleor checkout spa.   For visualization compare these two orders from the perspective of the dashboard, first the old experience:
![image](https://user-images.githubusercontent.com/479892/204067988-1a480384-4a3f-445b-956b-c41892fe6c2c.png)

Now the new experience:
![image](https://user-images.githubusercontent.com/479892/204068058-ab8ce3b0-effa-4ee0-b546-cfb0e9d70179.png)

Some key differences are:
1.  by default the new Stripe integration will capture the funds immediately, whereas the old integration would have manual payment intent where capture would be performed in the dashboard.  This is important because capturing incurs network fees and requires a refund to cancel, whereas an authorization can be voided without fees.   I was able to figure out how to pass the payment_intent of manual to Stripe to get the authorization without the capture as before.
2. No payment record is created, in fact I see minimal mutations for creating a payment in the `saleor-checkout-app`.   Everything seems to create transactions now, but there isn't a checkout or a payment record in the database.  It sees that the payment record is used by the dashboard to allow the order payment to be captured or voided through the admin, but now with the new experience there is a disconnect between the dashboard and Stripe.  It would be possible to manage captures in Stripe and mark the order as paid in the dashboard, but there version I have now from ~May of this year does not have this disconnect.
3.  The order confirmed email gets sent as soon as the customer clicks on pay to head to the Stripe checkout experience whereas before the inventory was not deducted and the order not confirmed until the payment was successfull.  

I've been spending some hours trying to think through how to create the payment record from within the stripe callback in  `saleor-checkout-app`, but it seems to be that nothing makes the checkoutId available to this app -- there is an orderId but the `MutationCheckoutPaymentCreateArgs` wants a checkoutId and not an orderId, so I am kind of at a loss for how to create the payment record and restore the stripe payment actions in the dashboard to capture or void an authorized payment.   This is a blocker for me to be able to roll out the updated saleor and so I would love some additional guidance with respect to this. 

I like the UI and checkout experience for the user better in this new flow, but as the store administrator this makes my work of managing the store and its inventory more complicated.

**Reply by timuric** [reply]
Replied here https://github.com/saleor/react-storefront/discussions/686
